schemas/da00_dataarray.fbs

```diff
@@ -0,0 +1,26 @@
+
+// A flatbuffer schema for holding histogram data or EPICS area detector updates
+// A merger of hs00/hs01 and ADAr aimed at compatibility with scipp DataArrays
+
+file_identifier "da00";
+
+enum da00_dtype: byte {none, int8, uint8, int16, uint16, int32, uint32, int64, uint64, float32, float64, c_string}
```

Comment:
@danesss commented on the previous errant PR https://github.com/ess-dmsc/streaming-data-types/pull/92#discussion_r1466072455
>This method for supporting several datatypes has the disadvantage that type conversion has to be done by the data consumer after decoding the flatbuffer.
>
>In other schemas (e.g. hs00, f144) we use a "union of tables" to achieve the same goal, with the advantage that the type safety is then ensured at the flatbuffer level instead of doing it in our code.
>
>I'm not sure we have actually observed simpler code when using the approach, but it should theoretically simplify the logic in our side.
Just a thought worth considering.

I am open to the union of tables idea, but I couldn't immediately see how to include "string-element"-valued (e.g., arbitrary `char`) data with the same flexibility.
At the moment a `da00_Variable` can be a multi-dimensional array _or_  (effectively) a fixed-length `char[]`. This lets it be used for either a `dataset` or an `attribute` in an HDF5 file.

@danesss, do you have an idea on how to achieve this? Or would you suggest splitting `dataset` and `attribute` tables, e.g.,

```
table da00_attribute {
  name: string (required);
  value: string (required);
  ...
}

table da00_Dataset {
  name: string (required);
  unit: string;
  label: string;
  source: string;
  axes: [string] (required);
  shape [long] (required);
  data: {UnionDataTableType};
}

table da00_DataArray {
  source_name: string (required);
  timestamp: ulong;
  datasets: [da00_Dataset];
  attributes: [da00_Attributes];
}
```